packages/contracts-bedrock/src/L2/L2StandardBridge.sol

```diff
@@ -52,20 +52,24 @@ contract L2StandardBridge is StandardBridge, ISemver {
         bytes extraData
     );
 
-    /// @custom:semver 1.7.0
-    string public constant version = "1.7.0";
+    /// @custom:semver 1.8.0
+    string public constant version = "1.8.0";
 
     /// @notice Constructs the L2StandardBridge contract.
     /// @param _otherBridge Address of the L1StandardBridge.
-    constructor(address payable _otherBridge)
-        StandardBridge(payable(Predeploys.L2_CROSS_DOMAIN_MESSENGER), _otherBridge)
-    {
-        initialize();
+    constructor(address payable _otherBridge) StandardBridge() {
+        initialize({
+            _messenger: CrossDomainMessenger(payable(Predeploys.L2_CROSS_DOMAIN_MESSENGER)),
+            _otherBridge: StandardBridge(_otherBridge)
+        });
     }
 
-    /// @notice Initializes the contract. This is a noop in the implementation but included to ensure that
-    ///         the contract cannot be initialized a second time.
-    function initialize() public initializer { }
+    /// @notice Initializer.
+    /// @param _messenger Contract for the CrossDomainMessenger on this chain.
+    /// @param _otherBridge Contract for the corresponding bridge on the other chain.
+    function initialize(CrossDomainMessenger _messenger, StandardBridge _otherBridge) public initializer {
```

Comment:
We always know the address of the `CrossDomainMessenger` on L2 so we can hardcode the value passed to `__StandardBridge_init` to the predeploy constant